src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -149,15 +149,123 @@ ucs_status_t uct_rdmacm_cm_ep_conn_notify(uct_ep_h ep)
     return cep->status;
 }
 
+/**
+ * Allocate a reserved QPN either from the last FW object allocated,
+ * or by allocating a new one. When find a free QPN in an object, it
+ * always starts from last allocation position, to make sure the QPN
+ * always move forward to prevent stale QPN.
+ */
+static ucs_status_t
+uct_rdamcm_cm_ep_create_reserved_qpn(uct_rdmacm_cm_ep_t *cep,
+                                     struct uct_rdmacm_cm_qp_context *ctx)
+{
+    ucs_status_t status = UCS_OK;
+    struct uct_rdmacm_cm_reserved_qpn_blk *blk;
+    uint32_t qpns_per_obj;
+
+    qpns_per_obj = uct_rdmacm_cm_get_qpns_per_obj(ctx->reserved_qpn_ctx);
```

Comment:
init inline

Reply:
@yosefe Do you mean change as following?


<!--StartFragment-->    uint32_t qpns_per_obj = uct_rdmacm_cm_get_qpns_per_obj(ctx-&gt;reserved_qpn_ctx);<!--EndFragment-->

Reply:
yes

Reply:
@yosefe Fixed.